src/MizarData/include/MizarData/FrameTrack.h

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef MIZAR_DATA_MIZAR_FRAME_TRACK_H_
+#define MIZAR_DATA_MIZAR_FRAME_TRACK_H_
+
+#include "ClientData/ScopeId.h"
+#include "ClientData/ScopeInfo.h"
+#include "GrpcProtos/capture.pb.h"
+#include "OrbitBase/Typedef.h"
+#include "stdint.h"
+
+namespace orbit_mizar_data {
+
+struct FrameTrackIdTag {};
+// The type is used identify a frame track in Mizar. It can be either ETW or a scope.
+using FrameTrackId =
+    orbit_base::Typedef<FrameTrackIdTag, std::variant<orbit_client_data::ScopeId,
+                                                      orbit_grpc_protos::PresentEvent::Source>>;
+
+struct FrameStartNsTag {};
+// Absolute timestamp of the capture start in nanos
+using FrameStartNs = orbit_base::Typedef<FrameStartNsTag, uint64_t>;
```

Comment:
I've already spent some time more than once to make sure the frame is defined by two starts, not by two ends. So I think the type is justified here. 

> This leads to duplicate template instantiations where we don't expect it. std::vector<MyTaggedUInt64_1> is different than std::vector<MyTaggedUInt64_2> 

I agree, yet I rather see it as an advantage. Why would you move between two vectors of independent types? 

> See answer to question about tagged integers...

We surely don't pay more memory (I can add a `static_assert` if you wish), we shouldn't pay more runtime that we pay using `unique_ptr` instead of a raw ptr (in fact, I think, most of the overhead (if not all of it) will be optimised away). 

About the compile-time. I already use `Typedefs` quite a bit in the project. For instance, `ScopeId` (the one replacing `uint64_t scope_id` that in turn generalises `uint64_t function_id`) is used in numerous places across the codebase and it doesn't seem to significantly increase the compile type. And even if the compile time is increased I prefer to pay with machine time rather than with the reader's time. Machines should work; People should think.

In fact, I think the whole way Mizar (and Orbit to a large extent) treats timestamps should be refactored in the future via using `std::chrono`.

Reply:
> Why would you move between two vectors of independent types?

Well they are not truly independent, that's my point. System A's "end" timestamps could be system B's "start" timestamps. If system B asks for A's "end" timestamps so that it can use them as its own "start" timestamps, then a type conversion needs to happen instead of a trivial move. Anyway, not a blocker, this is Mizar specific. But it might be worth discussing with the team where we should draw the line.

> And even if the compile time is increased I prefer to pay with machine time rather than with the reader's time. Machines should work; People should think.

I definitely see compile time and quick iteration as something that we should optimize. But yeah, it's all about tradeoffs.